server/src/app/auth/auth.service.spec.ts

```diff
@@ -0,0 +1,220 @@
+        ],
+      }).compile();
+
+      service = module.get<AuthService>(AuthService);
+      expect(service).toBeDefined();
+    });
+  });
+
+  describe('#createToken()', () => {
+    it('should return a valid token', async () => {
+      const result = {
+        expiresIn: 3600,
+        accessToken: 'address',
+      };
+
+      when(mockJwtService.sign(anything())).thenReturn(result.accessToken);
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      expect(service.createToken('address')).toEqual(result);
+    });
+
+    it('should throw "address should be defined"', async () => {
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      expect(() => {
+        service.createToken(undefined);
+      }).toThrow('address should be defined.');
+    });
+  });
+
+  describe('#signUp', () => {
+    it('should create a user', async () => {
+      const mockService: AuthService = mock(AuthService);
+
+      const user: User = new User(address, name);
+      const jwt: Token = {
+        expiresIn: 3600,
+        accessToken: 'token',
+      };
+
+      const signUpResponseDto: SignUpResponseDto
+        = new SignUpResponseDto(accessToken, address, name);
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(true);
+
+      when(mockUserService.create(address, name))
+        .thenReturn(new Promise((resolve) => {
+          resolve(user);
+        }));
+
+      when(mockService.createToken(address))
+        .thenReturn(jwt);
+
+      when(mockJwtService.sign(objectContaining({
+        expiresIn: 3600,
+        address,
+      }))).thenReturn(accessToken);
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature)).resolves.toEqual(signUpResponseDto);
+    });
+    it('should throw UnauthorizedException()', async () => {
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(false);
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
+    });
+    it('should throw UnauthorizedException()', async () => {
+      const mockService: AuthService = mock(AuthService);
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(true);
+
+      when(mockUserService.create(address, name))
+        .thenReturn(new Promise((resolve) => {
+          resolve(undefined);
+        }));
+
+      when(mockService.createToken(address))
+        .thenReturn(undefined);
+
+      when(mockJwtService.sign(address)).thenReturn(undefined);
```

Comment:
does mockJwtService return undefined because it is signing without expireIn? 

then it should be InvalidParameterError, not UnauthorizedError.

and i think u don't have to test this case
1. jwtService is used at createToken, not signUp.
2. there is no case to put invalid parameter on jwtService now.